services/src/server.rs

```diff
@@ -193,83 +165,3 @@ where
     .await
     .map_err(Into::into)
 }
-
-#[cfg(test)]
-mod tests {
-    use url::Url;
-
-    use super::*;
-    use crate::contexts::{Session, SimpleSession};
-    use crate::handlers::ErrorResponse;
-
-    /// Test the webserver startup to ensure that `tokio` and `actix` are working properly
-    #[actix_rt::test]
-    async fn webserver_start() {
-        tokio::select! {
-            server = start_server(None) => {
-                server.expect("server run");
-            }
-            _ = queries() => {}
-        }
-    }
-
-    async fn queries() {
-        let web_config: config::Web = get_config_element().unwrap();
-        let base_url = Url::parse(&format!(
-            "http://{}{}/",
-            web_config.bind_address, web_config.api_prefix
-        ))
-        .unwrap();
-
-        assert!(wait_for_server(&base_url).await);
-        issue_queries(&base_url).await;
-    }
-
-    async fn issue_queries(base_url: &Url) {
-        let client = reqwest::Client::new();
-
-        let body = client
-            .post(base_url.join("anonymous").unwrap())
-            .send()
-            .await
-            .unwrap()
-            .text()
-            .await
-            .unwrap();
-
-        let session: SimpleSession = serde_json::from_str(&body).unwrap();
-
-        let body = client
-            .post(base_url.join("project").unwrap())
-            .header("Authorization", format!("Bearer {}", session.id()))
-            .header("Content-Type", "application/json")
-            .body("no json")
-            .send()
-            .await
-            .unwrap()
-            .text()
-            .await
-            .unwrap();
-
-        assert_eq!(
-            serde_json::from_str::<ErrorResponse>(&body).unwrap(),
-            ErrorResponse {
-                error: "BodyDeserializeError".to_string(),
-                message: "expected ident at line 1 column 2".to_string()
-            }
-        );
-    }
-
-    const WAIT_SERVER_RETRIES: i32 = 5;
-    const WAIT_SERVER_RETRY_INTERVAL: u64 = 1;
-
-    async fn wait_for_server(base_url: &Url) -> bool {
-        for _ in 0..WAIT_SERVER_RETRIES {
-            if reqwest::get(base_url.clone()).await.is_ok() {
-                return true;
-            }
-            std::thread::sleep(std::time::Duration::from_secs(WAIT_SERVER_RETRY_INTERVAL));
-        }
-        false
-    }
-}
```

Comment:
why are the tests gone?

Reply:
IMHO these should be integration tests, and not unit tests. I didn't find a way to make these work with the temp context, so instead, I extended the services/tests/startup.rs test to no only check for warnings, but also to check whether the webserver can be connected after startup.